### PR TITLE
DC-6953 Add two-way-message renderer endpoint

### PR DESCRIPTION
**Pull request by kjainsc**

### CODE REVIEW

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala [hunk 1]
```diff
@@ -53,4 +55,34 @@ class SecureMessageRenderer @Inject() (
       case r => InternalServerError
     }
   }
+
+  def getContentBy(id: String, msgType: String): Action[AnyContent] = Action.async { implicit request =>
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+    def createMsg(replyType: RenderType.ReplyType): Future[Result] =
```
Comment (satyajosyula2020) [inline app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala:62]: Is it actually creating message or rendering the message? Better to have right name

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala [hunk 2]
```diff
@@ -53,4 +55,34 @@ class SecureMessageRenderer @Inject() (
...
+      messageService.findMessageListById(id).flatMap {
+        case Right(msgList) => getHtmlResponse(id, msgList, replyType)
+        case Left(err) =>
+          logger.warn(s"Error retrieving messages: $err")
+          Future.successful(BadGateway(err))
```
[the remark this answers is not in the document]
Comment (ardwalker) [reply app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala:67]: Could be good to return whatever the relevant Left side is that's returned by findMessageList. Can make it a bit easier to troubleshoot.

File: app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    val fullConversation = conversation.mkString(Xhtml.toXhtml(<hr/>))
+
+    Future.successful(Right(Html.apply(fullConversation)))
+  }
+
+  private def createConversationList(messages: List[ConversationItem], replyType: RenderType.ReplyType): List[String] =
+    replyType match {
+      case RenderType.CustomerLink =>
+        messages
+          .sortWith(_.id > _.id)
```
Comment (satyajosyula2020) [inline app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala:53]: Looks like this sorting is redundant, you are already doing that before calling this method itself. Also, the code in the 2 cases is almost same expect hasSmallSubject. So, there is a scope of removing duplicated code for better clarity.

Comment (satyajosyula2020) [reply app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala:53]: Ignore my comment if its lift and shift, as you can refactor it later.

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala (hunk 1 above)
Comment (kjainsc) [reply app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala:62]: updated now, thanks

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjainsc) [reply app/uk/gov/hmrc/securemessage/controllers/SecureMessageRenderer.scala:67]: to be refactored separately, thanks

File: app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala (hunk 3 above)
Comment (kjainsc) [reply app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala:53]: yes, will refactor it separately. thanks

File: app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    }
+
+  // format: off
+  private def format2wsMessageForCustomer(item: ConversationItem, metadata: ItemMetadata): String =
+    Xhtml.toXhtml(
+      getHeader(metadata, item.subject) ++ <p class="faded-text--small">{getCustomerDateText(item)}</p>
+        ++ getContentDiv(item.content) ++
+        item.body
+          .flatMap(_.enquiryType)
+          .map(_.startsWith("p800"))
```
Comment (ardwalker) [inline app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala:83]: Is this intentional - p800?

Comment (kjainsc) [reply app/uk/gov/hmrc/securemessage/services/HtmlCreatorService.scala:83]: yes

**Review by ardwalker — APPROVED** [review]

**Pull request merged**